test/common/testlib.py

```diff
@@ -767,12 +779,42 @@ def snapshot(self, title, label=None):
             attach(filename, move=True)
             print("Wrote HTML dump to " + filename)
 
-    def assert_pixels(self, selector, key, ignore=[]):
-        """Compare the given element with its reference"""
+    def _set_window_size(self, width, height):
+        self.cdp.invoke("Emulation.setDeviceMetricsOverride",
+                        width=width, height=height,
+                        deviceScaleFactor=0, mobile=False)
+
+    def set_layout(self, name):
+        layout = next(filter(lambda l: l["name"] == name, self.layouts))
+        if layout and layout != self.current_layout:
```

Comment:
`if layout` is always true, as the lookup throws an exception. (But probably moot -- let's just use a proper dict)

Reply:
Good point.